lib/spack/spack/cmd/env.py

```diff
@@ -142,14 +144,16 @@ def env_deactivate_setup_parser(subparser):
 def env_deactivate(args):
     if not args.shell:
         msg = [
-            "This command works best with Spack's shell support",
+            "`spack env deactivate` works best with spack's shell support.",
```

Comment:
There's so much in common between the `activate` and `deactivate` messages that it seems like this could be factored out and passed the subcommand to reduce redundancy and ensure consistency.